Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java

```diff
@@ -305,56 +305,42 @@ public final void disconnect() {
     }
 
     /**
-     * Runs both the update Start end End position functions. Use when
-     * refreshing UI representation of the Line.
+     * Removes this Connection, disconnecting its anchors and removing this Connection from the pane it is on.
      */
-    public void invalidateAnchorPositions() {
-        startAnchor.ifPresent(a -> setStartPosition(a.getCenterInPane()));
-        endAnchor.ifPresent(a -> setEndPosition(a.getCenterInPane()));
+    public final void remove() {
+        disconnect();
+        getPane().getChildren().remove(this);
     }
-    
+
     /**
-     * Tells the Connection that its current state (considering connections) possibly has
-     * changed.
-     *
-     * This method should only be called after the Connection's constructor is done.
-     * 
-     * This method will invalidate the Connection even if the state did not change.
+     * Adds the listeners this Connections needs to keep its visual
+     * representation up-to-date to the given anchor.
      */
-    private void invalidateConnectionState() {
-        invalidateAnchorPositions();
-        checkError();
+    private void addListeners(ConnectionAnchor anchor) {
+        anchor.localToSceneTransformProperty().addListener(this);
```

Comment:
Did not do that because I was unsure of being able to remove the method (wether or not referencing a method twice is an equal reference)